promql/engine.go

```diff
@@ -833,10 +844,14 @@ func (ng *Engine) populateSeries(querier storage.Querier, s *parser.EvalStmt) {
 		switch n := node.(type) {
 		case *parser.VectorSelector:
 			start, end := ng.getTimeRangesForSelector(s, n, path, evalRange)
+			interval := s.Interval
+			if interval == 0 {
```

Comment:
Should you not always try to get the subquery interval and set that if it is non-zero? Because that is what will be applied to the vector selector in that case. Can you try a query on Thanos with query interval and subquery interval are incompatible (i.e. not divisible by each other).

Reply:
s.Interval will be 0 only in the case of an instant query I believe

Reply:
Even in case of a range query, the samples required for a subquery will be aligned with the subquery interval and not the range query interval (think of subquery as yet another range query with some timestamp alignments). So I meant literally only taking innermost subquery interval iff it exists. Was there any specific reason to prefer the range query interval if there was a subquery?

Reply:
I did not know that. Let me update my branch.